Cython/Build/Tests/TestRecythonize.py

```diff
@@ -1,212 +1,197 @@
+        # Change a.c
+        self.fresh_cythonize(module)
 
+        definition_after = self.relative_lines_from_file(
+            module_c_file, module_line_1, 0, 7)
 
-        # Cythonize to create b.c
-        fresh_cythonize([a_pyx, b_pyx])
+        self.assertNotIn("a_x = 1;", definition_after, SAME)
+        self.assertIn("a_x = 1.0;", definition_after, INCORRECT)
 
-        # Sleep to address coarse time-stamp precision.
-        time.sleep(1)
+    # pxd_exists_for_first_check is not used because cimport requires pxd
+    # to import another script.
+    def recythonize_on_dep_pxd_change(self, ext_a, ext_b):
+        dep_filename, module_filename = "a" + ext_a, "b" + ext_b
 
-        with open(b_c) as f:
-            b_c_contents1 = f.read()
+        pxd_to_be_modified = os.path.join(self.src_dir, 'a.pxd')
+        module_dependency = os.path.join(self.src_dir, dep_filename)
+        dep_c_file = os.path.join(self.src_dir, 'a.c')  # should change
+        pxd_for_cimport = os.path.join(self.src_dir, 'b.pxd')
+        module = os.path.join(self.src_dir, module_filename)
+        module_c_file = os.path.join(self.src_dir, 'b.c')  # should change
 
-        with open(a_pxd, 'w') as f:
-            f.write('cdef double value\n')
+        dep_line_1 = LINE_BEFORE_IMPLEMENTATION.format(
+            filename=dep_filename, at_line=1)
+        module_line_1 = LINE_BEFORE_IMPLEMENTATION.format(
+            filename=module_filename, at_line=1)
 
-        fresh_cythonize([a_pyx, b_pyx])
+        write_file(pxd_to_be_modified, 'cdef int x\n')
+        write_file(module_dependency, 'x = 1\n')
+        write_file(pxd_for_cimport, 'cimport a\n')
+        write_file(module, 'a.x = 2\n')
+
+        dependencies = self.fresh_all_dependencies(module)
+        self.assertIn(pxd_for_cimport, dependencies)
+        self.assertIn(module, dependencies)
+        self.assertIn(pxd_to_be_modified, dependencies)
+        self.assertEqual(3, len(dependencies))
 
-        with open(b_c) as f:
-            b_c_contents2 = f.read()
+        # Create a.c and b.c
+        self.fresh_cythonize([module_dependency, module])
 
+        dep_definition_before = self.relative_lines_from_file(
+            dep_c_file, dep_line_1, 0, 7)
+
+        module_definition_before = self.relative_lines_from_file(
+            module_c_file, module_line_1, 0, 7)
 
+        self.assertIn("a_x = 1;", dep_definition_before, INCORRECT)
+        self.assertIn("a_x = 2;", module_definition_before, INCORRECT)
 
-        self.assertTrue("__pyx_v_1a_value = 2;" in b_c_contents1)
-        self.assertFalse("__pyx_v_1a_value = 2;" in b_c_contents2)
-        self.assertTrue("__pyx_v_1a_value = 2.0;" in b_c_contents2)
-        self.assertFalse("__pyx_v_1a_value = 2.0;" in b_c_contents1)
+        # See https://github.com/cython/cython/issues/4245
+        write_newer_file(pxd_to_be_modified, pxd_to_be_modified, 'cdef float x\n')
```

Comment:
This seems wrong. I think `newer_than` should refer to the `.c` file, not the `.pxd` file. (Probably also applies to other places.)

Reply:
resolved [here](https://github.com/cython/cython/pull/4241/commits/07c56aa7c5101fc677d82edeed86c083993d12d8)